src/System.Management.Automation/CoreCLR/CorePsPlatform.cs

```diff
@@ -18,6 +18,13 @@ public static class Platform
     {
         private static string _tempDirectory = null;
 
+#if UNIX
+        // Cache the path so that GetTemporaryDirectory() does not create
+        // a temp directory for every new runspace if HOME environment variable is not present or empty.
+        private static string s_envHome =
+            System.Environment.GetEnvironmentVariable(CommonEnvVariableNames.Home) ?? GetTemporaryDirectory();
```

Comment:
If we cache `HOME`, then why not cache other values read from environment variables like `XDG_CONFIG_HOME`? Also we [don't cache the "MyDocument" path on Windows](https://github.com/PowerShell/PowerShell/blob/master/src/System.Management.Automation/engine/Modules/ModuleIntrinsics.cs#L970). I think just caching a temp directory path on the first use would be sufficient.

Reply:
It seems it is not on hot path, I do not think we get a win.

Do we continue your PR or the PR? :-)

Reply:
My PR already did that and it's only 2 lines change 😄 How about just close this PR?